exporters/cr/exporter.go

```diff
@@ -0,0 +1,142 @@
+package cr
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8srestconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+var certManagerCertificateGroupVersionResource = schema.GroupVersionResource{
+	Group:    "cert-manager.io",
+	Resource: "certificates",
+	Version:  "v1",
+}
+
+type Config struct {
+	Namespaces []string
+}
+
+type Exporter struct {
+	certNotAfter  *prometheus.Desc
+	ctx           context.Context
+	logger        micrologger.Logger
+	dynamicClient dynamic.Interface
+
+	namespaces []string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Namespaces: []string{},
+	}
+}
+
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.logger.Log("info", "start collecting metrics")
+
+	namespacesToCheck := []string{""}
+	// Create a list of namespaces to check.
+	if len(e.namespaces) != 0 {
+		namespacesToCheck = e.namespaces
+	}
+
+	listOptions := metav1.ListOptions{}
+
+	// Loop over namespaces.
+	for _, namespace := range namespacesToCheck {
+		certs, err := e.dynamicClient.Resource(certManagerCertificateGroupVersionResource).Namespace(namespace).List(e.ctx, listOptions)
```

Comment:
@ubergesundheit has this change also been tested on WCs? It's currently failing for me with the following error:
```
{"caller":"github.com/giantswarm/cert-exporter/exporters/cr/exporter.go:59","error":{"kind":"unknown",
"annotation":"the server could not find the requested resource","stack":[{"file":"/root/project/export
ers/cr/exporter.go","line":59}]},"time":"2021-09-02T11:00:28.011879+00:00"}
```
because it's trying to list all `certificates.cert-manager.io` and that CRD does not exist on WCs.

Reply:
Yeah tested on clusters without cert-manager installed. Does it crash completely or is it just printing the error?

Reply:
It's just printing the error and continuing without doing anything. So it's an expected behaviour if there is no `cert-manager`? If that's the case maybe it makes sense to add an additional check and not print the error.

Reply:
It should export the metric nevertheless. Checked again on the cluster and everything works fine. :+1: 

I like your suggestion of disabling the error. I'll need to revisit cert-exporter pretty soon anyway, I think I'll add some code hiding this if cert-manager is not available.